### ISSUE TITLE
DenseTensor?

### ISSUE THREAD

**Issue by Hydrotoast** [scalanlp/breeze#590]
From the wiki, I see that there exists `DenseVector` and `DenseMatrix`; however, is there a `DenseTensor` data structure?

**Comment by dlwh** [comment]
no. people have offered to add one about 4-5 times, but no one's followed
through yet.

**Comment by stormy-ua** [comment]
+1 to the issue. I was thinking about building a simple deep learning algorithms with breeze, but the lack of the support for Tensor makes it infeasible.
